docs/DeveloperGuide.md

```diff
@@ -452,6 +462,148 @@ facilitates the lending of money and ensures data integrity by validating input
+- *isBefore*: Determines whether an input DateTime object has a dateTime field that is before that of the current
+instance.
+- *isAfter*: Determines whether an input DateTime object has a dateTime field that is after that of the current
+instance.
+- *isEqual*: Determines whether an input DateTime object has a dateTime field that is equal to that of the current 
+instance.
+- *isFuture*: Determines whether the dateTime field of the current instance represent a future date & time (e.g. is 
+after the preset system time). Currently used within the constructor only.
+- *toStorageString*: Formats the dateTime field of the current instance into a String output suitable for loading and 
+storing.
+- *toString*: Overrides the default toString method. Formats the dateTime field of the existing DateTime instance into a
+String output suitable for printouts.
+
+<ins>Usage Example</ins>
+
+The following UML diagram displays how the dateTime component is handled when the user is adding a dated transaction.
+
+![addDateTimeforDatedTransaction.png](diagrams%2FaddDateTimeforDatedTransaction.png)
+
+Given below is an example usage scenario of how the DateTime class behaves at each step in adding dated transactions:
+
+1. Following steps 1-3 of the scenario of adding a new transaction, the Transaction class now identifies a potential 
+presence of a dateTime component in the userExpression through the specified prefix.
+2. It initiates the constructor method of the DateTime class and attempts to create a new object to store the user input 
+date & time. Validation of the dateTimeExpression will occur in the DateTime class at this stage.
+3. If the dateTimeExpression from the user is valid, the corresponding DateTime object will be returned to the
+Transaction class as a result and stored as the dateTime of the transaction.
+4. If the dateTimeExpression from the user is in the wrong format, exception occurs and the DateTime class will output 
+the "Invalid dateTime format" warning through the logger.
+5. If the dateTimeExpression from the user is an unrealistic future date & time, exception occurs and the DateTime class
+will output the "Invalid dateTime input" warning through the logger.
+6. If the dateTime component is appended successfully, the Transaction class will proceed to handle other details of the
+transaction input, as per adding a normal transaction.
+
+The following Code Snippet outlines the above usage:
+```
+//In pareTransaction() method of the Transaction Class 
+if (splitInput[0].contains("t/")) { //Checks for the special prefix of date & time component while adding parsing user expression
+  String[] splitLenderTime = splitInput[0].split("t/", 2);
+  ...
+  this.transactionTime = new DateTime(splitLenderTime[1]);
+}
+```
+
+The following UML diagram displays how the dateTime component is handled when printout requests are initiated for dated 
+transactions.
+
+![printingDateTime](diagrams%2FprintingDateTime.png)
```

Comment:
Similar to above, drop the activation box. Current diagram with it is incorrect as activation box does not end. Also, class headers require a preceding colon, i.e. `:Transaction` as opposed to `Transaction`. This can be done using `":Transaction"` to parse in the colon as part of the class name.

Reply:
Resolved!